src/main/java/org/apache/pulsar/io/sftp/sink/MessageToRawFileOBSWriter.java

```diff
@@ -0,0 +1,95 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.io.sftp.sink;
+
+import com.obs.services.ObsClient;
+import com.obs.services.ObsConfiguration;
+import com.obs.services.model.PutObjectRequest;
+import com.obs.services.model.PutObjectResult;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.security.NoSuchAlgorithmException;
+import java.util.Objects;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pulsar.client.api.Message;
+import org.apache.pulsar.functions.api.Record;
+import org.apache.pulsar.io.sftp.common.TaskState;
+import org.apache.pulsar.io.sftp.utils.Constants;
+import org.apache.pulsar.io.sftp.utils.FileUtil;
+import org.apache.pulsar.io.sftp.utils.HWObsUtil;
+
+/**
+ * Write files to local disk in parquet format.
+ */
+@Slf4j
+public class MessageToRawFileOBSWriter implements MessageOBSWriter<byte[]> {
+
+    @Override
+    public void writeToStorage(Record<byte[]> record, OBSSink obsSink, ObsConfiguration conf) {
+
+        OBSSinkConfig sinkConfig = obsSink.getOBSSinkConfig();
+        String outDirectory = sinkConfig.getOutDirectory();
+        RandomAccessFile randomFile = null;
+        try {
+            Message<byte[]> msg = record.getMessage().get();
+            byte[] contents = msg.getValue();
+            String name = new File(msg.getProperty(Constants.FILE_NAME)).getName();
+            String sftpPath = msg.getProperty(Constants.FILE_ABSOLUTE_PATH);
+            String fileName = StringUtils.isNotBlank(sftpPath)
+                    ? outDirectory.replaceFirst("/", "") + "/" + sftpPath + "/" + name
+                    : outDirectory.replaceFirst("/", "") + "/" + name;
+            String originalMD5 = msg.getProperty(Constants.FILE_MD5);
+            ObsClient obsClient = HWObsUtil.getObsClient(sinkConfig.getAccessKey(), sinkConfig.getSecretKey(),
+                    sinkConfig.getSecurityToken(), conf);
+            PutObjectRequest request = new PutObjectRequest();
+            request.setInput(new ByteArrayInputStream(contents));
+            request.setBucketName(sinkConfig.getBucket());
+            request.setObjectKey(fileName);
+            request.setExpires(sinkConfig.getExpires());
+            PutObjectResult result = obsClient.putObject(request);
```

Comment:
You can first determine whether the file exists and whether the MD5 is consistent. If it is consistent, you do not need to upload it again.